### PR TITLE
Update go client samples after release [API-1813]

### PR DESCRIPTION
**Pull request by fatihozer0**

### CODE REVIEW

**Comment by yuce** [comment]
@fatihozer0 Could you check that the new example works?

**Comment by Serdaro** [comment]
@fatihozer0 , I cannot validate if it works. Could you please let me know?

File: docs/modules/getting-started/pages/get-started-binary.adoc [hunk 1]
```diff
@@ -637,14 +638,14 @@ Go::
...
 
-  client, err := hazelcast.StartNewClientWithConfig(cb) 
+  ctx := context.TODO()
+  client, err := hazelcast.StartNewClientWithConfig(ctx, config)
 
   <1>
-  ctx := context.TODO()
   myMap, err := client.GetMap(ctx, "my-distributed-map")
   if err != nil {
     // handle the error
```
Comment (mehmettokgoz) [inline docs/modules/getting-started/pages/get-started-binary.adoc:651]: We should either `panic(err)` inside if or remove these handlers and add `// error handling is omitted for brevity` warning after line 640.

**Comment by yuce** [comment]
@fatihozer0 Could you try running the Go code in this PR and verify that it compiles / works OK?

**Review by yuce — CHANGES_REQUESTED** [review]

File: docs/modules/getting-started/pages/get-started-binary.adoc [hunk 2]
```diff
@@ -637,18 +639,21 @@ Go::
...
 
 func mapSampleRun() {
-  cb := hazelcast.NewConfigBuilder()
+  config := hazelcast.Config{}
 
-  cb.Cluster().SetName("hello-world")
+  config.Cluster.Name = "hello-world"
 
-  client, err := hazelcast.StartNewClientWithConfig(cb) 
+  ctx := context.TODO()
```
[the remark this answers is not in the document]
Comment (fatihozer0) [reply docs/modules/getting-started/pages/get-started-binary.adoc:653]: Done.

**Review by yuce — APPROVED** [review]
Looks good!

**Pull request merged**